### PR TITLE
Added Large File Uploader Wizard #7227

### PR DESCRIPTION
**Pull request by PritishC**
`TODO`: 

* Add tests for the wizard
* S3 related fields on `StaticFolder` should be invisible if the folder is not a S3 bucket
* Upload button view placement

### CODE REVIEW

File: view/static_folder_form.xml [hunk 1]
```diff
@@ -14,6 +16,8 @@
             <field name="s3_bucket_name"/>
             <label name="s3_object_prefix"/>
             <field name="s3_object_prefix"/>
+            <label name="s3_upload_from_ttl"/>
```
Comment (sharoonthomas) [inline view/static_folder_form.xml]: form_ttl

File: static_file.py [hunk 2]
```diff
@@ -91,6 +112,28 @@ class NereidStaticFile:
         fields.Boolean("S3 Bucket?"), 'get_is_s3_bucket'
     )
     s3_key = fields.Function(fields.Char("S3 key"), "get_s3_key")
+    is_large_file = fields.Boolean('Is Large File')
```
Comment (sharoonthomas) [inline static_file.py]: readonly=True

File: static_file.py [hunk 3]
```diff
@@ -207,4 +267,34 @@ def __setup__(cls):
...
+    base_url = 'https://openlabs.github.io/s3uploader/v1/upload.html'
+
+    def do_start(self, action):
+        """
+        This method overrides the action url given in XML and inserts the url
+        in the action object. It then proceeds to return the action.
+        """
+        StaticFile = Pool().get('nereid.static.file')
+
+        static_file = StaticFile(Transaction().context.get('active_id'))
```
Comment (sharoonthomas) [inline static_file.py]: static_file.is_large_file = True
static_file.save()

File: static_file.py (hunk 2 above)
Comment (PritishC) [inline static_file.py]: Done

File: static_file.py (hunk 3 above)
Comment (PritishC) [inline static_file.py]: Done

File: view/static_folder_form.xml (hunk 1 above)
Comment (PritishC) [inline view/static_folder_form.xml]: Done

File: static_file.py [hunk 4]
```diff
@@ -122,7 +165,7 @@ def _set_file_binary(self, value):
         """
         if not value:
             return
-        if self.type == "s3":
+        if self.type == "s3" and not self.is_large_file:
```
Comment (sharoonthomas) [inline static_file.py]: this check should be inside or it will go to the obvious condition and save in the local fs.

Comment (PritishC) [inline static_file.py]: Done.

File: static_file.py [hunk 5]
```diff
@@ -7,13 +7,20 @@
     :copyright: (c) 2013-2015 by Openlabs Technologies & Consulting (P) Limited
     :license: BSD, see LICENSE for more details.
 """
-from trytond.model import fields
-from trytond.pyson import Eval, Bool
 from boto.s3 import connection
 from boto.s3 import key
-from trytond.pool import PoolMeta
+from boto import exception
+import base64
```
Comment (tarunbhardwaj) [inline static_file.py]: base64 and json should be on top

File: static_file.py [hunk 6]
```diff
@@ -47,6 +58,10 @@ def __setup__(cls):
         cls._error_messages.update({
             "invalid_cname": "Cloudfront CNAME with '/' at the end is not " +
             "allowed",
+            "not_s3_bucket": "The file's folder is not an S3 bucket",
```
Comment (tarunbhardwaj) [inline static_file.py]: not `a`

**Pull request merged**